### ISSUE TITLE
Query fails when using .in(...) with empty array

### ISSUE THREAD

**Issue by threema-danilo** [juanluispaz/ts-sql-query#75]
With the following test case:

```diff
--- a/src/examples/BetterSqlite3SynchronousExample.ts
+++ b/src/examples/BetterSqlite3SynchronousExample.ts
@@ -705,6 +705,11 @@ function main() {
             .executeInsert())
         assertEquals(companiesIds, [8, 9, 10])

+        const companyIdOne = sync(connection.selectFrom(tCompany).where(tCompany.id.in([1])).selectOneColumn(tCompany.id).executeSelectMany());
+        assertEquals(companyIdOne, [1]);
+        const companyIdEmpty = sync(connection.selectFrom(tCompany).where(tCompany.id.in([])).selectOneColumn(tCompany.id).executeSelectMany());
+        assertEquals(companyIdEmpty, []);
+
         const parentCompany = tCompany.as('parentCompany')

         const parentCompanies = connection.subSelectUsing(tCompany)
```

...the query fails for the empty array:

```
ChainedError: Invalid int value to send to the db:
    at SelectQueryBuilder.query (/home/danilo/Threema/external/ts-sql-query/src/queryBuilders/SelectQueryBuilder.ts:64:19)
    at SelectQueryBuilder.__executeSelectMany (/home/danilo/Threema/external/ts-sql-query/src/queryBuilders/SelectQueryBuilder.ts:147:14)
    at SelectQueryBuilder.executeSelectMany (/home/danilo/Threema/external/ts-sql-query/src/queryBuilders/SelectQueryBuilder.ts:183:21)
    at main (/home/danilo/Threema/external/ts-sql-query/src/examples/BetterSqlite3SynchronousExample.ts:710:124)
    at Object.<anonymous> (/home/danilo/Threema/external/ts-sql-query/src/examples/BetterSqlite3SynchronousExample.ts:887:5)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Module.m._compile (/home/danilo/Threema/external/ts-sql-query/node_modules/ts-node/src/index.ts:1618:23)
    at Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Object.require.extensions.<computed> [as .ts] (/home/danilo/Threema/external/ts-sql-query/node_modules/ts-node/src/index.ts:1621:12)
    at Module.load (node:internal/modules/cjs/loader:1004:32)
Caused By: Error: Invalid int value to send to the db:
    at DBConnection.transformValueToDB (/home/danilo/Threema/external/ts-sql-query/src/connections/AbstractConnection.ts:1006:23)
    ... 8 lines matching cause stack trace ...
    at SqliteSqlBuilder._buildSelect (/home/danilo/Threema/external/ts-sql-query/src/sqlBuilders/AbstractSqlBuilder.ts:610:29) {
  [cause]: Error: Invalid int value to send to the db:
      at DBConnection.transformValueToDB (/home/danilo/Threema/external/ts-sql-query/src/connections/AbstractConnection.ts:1006:23)
      at DBConnection.transformValueToDB (/home/danilo/Threema/external/ts-sql-query/src/connections/AbstractSqliteConnection.ts:363:22)
      at SqliteSqlBuilder._transformParamToDB (/home/danilo/Threema/external/ts-sql-query/src/sqlBuilders/AbstractSqlBuilder.ts:464:45)
      at SqliteSqlBuilder._appendSpreadValue (/home/danilo/Threema/external/ts-sql-query/src/sqlBuilders/AbstractSqlBuilder.ts:410:35)
      at SqliteSqlBuilder._in (/home/danilo/Threema/external/ts-sql-query/src/sqlBuilders/AbstractSqlBuilder.ts:2111:80)
      at SqlOperationInValueSource.__toSql (/home/danilo/Threema/external/ts-sql-query/src/internal/ValueSourceImpl.ts:1135:44)
      at SqlOperationInValueSource.__toSqlForCondition (/home/danilo/Threema/external/ts-sql-query/src/internal/ValueSourceImpl.ts:69:21)
      at SqliteSqlBuilder._appendCondition (/home/danilo/Threema/external/ts-sql-query/src/sqlBuilders/AbstractSqlBuilder.ts:367:30)
      at SqliteSqlBuilder._buildSelectWithColumnsInfo (/home/danilo/Threema/external/ts-sql-query/src/sqlBuilders/AbstractSqlBuilder.ts:849:41)
      at SqliteSqlBuilder._buildSelect (/home/danilo/Threema/external/ts-sql-query/src/sqlBuilders/AbstractSqlBuilder.ts:610:29)
}
```

I would expect a `WHERE id IN ()` to be generated, returning no rows.

Is this a bug in the library?

**Comment by threema-danilo** [comment]
My current workaround in my code is this hack 😄

```ts
const whereClause =
    fileDataUids.length === 0
        ? tFileData.uid.notEquals(tFileData.uid)
        : tFileData.uid.in(fileDataUids);
```

**Comment by juanluispaz** [comment]
Oh, some databases doesn't support in over empty array (like PostgreSQL); in that case I generate an alternative SQL (a false boolean) to allow it works as expected. I will include Sqlite on this.

**Comment by juanluispaz** [comment]
Ups, it is not that; Sqlite support in over an empty list; it is an internal bug generating the empty list. I will fix it.

**Comment by juanluispaz** [comment]
Fixed in ts-sql-query 1.48.0 already published

**Comment by threema-danilo** [comment]
Great, thanks @juanluispaz!
